eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/MockableResourceProvider.cs

```diff
@@ -148,7 +259,16 @@ static MethodProvider BuildGetMethod(TypeProvider enclosingType, MethodProvider
             }
         }
 
-        // TODO -- when we have the ability to get parent resources, we might move this to a more generic place and make it a helper method.
+        private MethodProvider BuildNonResourceMethod(NonResourceMethod method, bool isAsync)
+        {
+            var clientInfo = _clientInfos[method.InputClient];
+            return method.InputMethod switch
+            {
+                InputPagingServiceMethod pagingMethod => new PageableOperationMethodProvider(this, ContextualPath, clientInfo, pagingMethod, isAsync),
+                _ => new ResourceOperationMethodProvider(this, ContextualPath, clientInfo, method.InputMethod, isAsync)
```

Comment:
since this is non-resource method, why are we still constructing ResourceOperationMethodProvider?

Reply:
The fact that it works fine here indicates this class ResourceOperationMethodProvider is not properly named.
I like the concept of "contextual". The provider (resource/collection/mockable resource) is a container holding a contextual path, and methods on these providers would consume that contextual path.
In this way, all the methods in our generated libraries are this kind of "contextual" method, because all of them will be holding on such a class providing an "Id" as a contextual path.

On the other hand, the non-resource methods here are holding by mockable resource - therefore you could also consider the word `resource` in this class name is referring to the carrier of this method instead of saying this method is a resource method.